src/core/execute.c

```diff
@@ -2462,12 +2484,36 @@ static int setup_keyring(
                 else if (errno == EDQUOT)
                         log_unit_debug_errno(u, errno, "Out of kernel keyrings to allocate, ignoring.");
                 else
-                        return log_unit_error_errno(u, errno, "Setting up kernel keyring failed: %m");
+                        r = log_unit_error_errno(u, errno, "Setting up kernel keyring failed: %m");
 
-                return 0;
+                goto out;
```

Comment:
you should reset r to 0 here for the cases where we ignore the error... otherwise its value will be undefined

Reply:
actually, r was never set in all successful branches either, thus initialized it to zero at the very top of the function. Repushed.